src/mds/CInode.cc

```diff
@@ -3171,7 +3171,7 @@ void CInode::choose_lock_state(SimpleLock *lock, int allissued)
     if (lock->is_xlocked()) {
       // do nothing here
     } else if (lock->get_state() != LOCK_MIX) {
-      if (issued & (CEPH_CAP_GEXCL | CEPH_CAP_GBUFFER))
+      if (issued & (CEPH_CAP_GEXCL | CEPH_CAP_GBUFFER | CEPH_CAP_GCACHE))
```

Comment:
If filelock is in SYNC state, CEPH_CAP_GCACHE can be issued.  

The test should be:
 if (issued & CEPH_CAP_GWR) {
   if (issued & (CEPH_CAP_GCACHE | CEPH_CAP_GSHARE)) {
        lock->set_state(LOCK_EXCL);
      else
         lock->set_state(LOCK_MIX);
    }
}

Reply:
> If filelock is in SYNC state, CEPH_CAP_GCACHE can be issued.
> 
Yeah, right. I will fix it.


> The test should be:
> if (issued & CEPH_CAP_GWR) {
> if (issued & (CEPH_CAP_GCACHE | CEPH_CAP_GSHARE)) {
> lock->set_state(LOCK_EXCL);
> else
> lock->set_state(LOCK_MIX);
> }
> }